velox/connectors/hive/storage_adapters/s3fs/S3FileSystem.cpp

```diff
@@ -627,7 +627,9 @@ class S3FileSystem::Impl {
  public:
   Impl(const S3Config& s3Config) {
     VELOX_CHECK(getAwsInstance()->isInitialized(), "S3 is not initialized");
-    Aws::S3::S3ClientConfiguration clientConfig;
+    Aws::Client::ClientConfigurationInitValues initValues;
```

Comment:
The S3Client constructor with Aws::Client::ClientConfiguration argument is deprecated. Use `Aws::S3::S3ClientConfiguration`. I see there is an option to disable IMDS for that as well.